pyroomacoustics/room.py

```diff
@@ -2871,3 +2869,122 @@ def is_inside(self, pos):
+
+    Parameters
+    ----------
+    dim: int
+        Dimension of the room (2 or 3).
+    fs: int, optional
+        The sampling frequency in Hz. Default is 8000.
+    t0: float, optional
+        The global starting time of the simulation in seconds. Default is 0.
+    sigma2_awgn: float, optional
+        The variance of the additive white Gaussian noise added during
+        simulation. By default, none is added.
+    sources: list of SoundSource objects, optional
+        Sources to place in the room. Sources can be added after room creating
+        with the `add_source` method by providing coordinates.
+    mics: MicrophoneArray object, optional
+        The microphone array to place in the room. A single microphone or
+        microphone array can be added after room creation with the
+        `add_microphone_array` method.
+    temperature: float, optional
+        The air temperature in the room in degree Celsius. By default, set so
+        that speed of sound is 343 m/s.
+    humidity: float, optional
+        The relative humidity of the air in the room (between 0 and 100). By
+        default set to 0.
+    air_absorption: bool, optional
+        If set to True, absorption of sound energy by the air will be
+        simulated.
+    """
+
+    def __init__(
+        self,
+        dim,
+        fs=8000,
+        t0=0.0,
+        sigma2_awgn=None,
+        sources=None,
+        mics=None,
+        temperature=None,
+        humidity=None,
+        air_absorption=False,
+    ):
+        if not dim in [2, 3]:
+            raise ValueError("Anechoic room dimension has to be either 2 or 3.")
+
+        # Setting max_order to 0 emulates an anechoic room.
+        max_order = 0
+
+        # Ray tracing only makes sense in echoic rooms.
+        ray_tracing = False
+
+        # Create some dummy walls
+        p = np.ones((dim,))
+
+        # The materials are not actually used because max_order is set to 0 and ray-tracing to False.
+        # Anyways, we use the energy_absorption and scattering corresponding to an anechoic room.
+        materials = Material(energy_absorption=1.0, scattering=0.0)
+
+        # Set deprecated parameter
+        absorption = None
+
+        ShoeBox.__init__(
+            self,
+            p=p,
+            fs=fs,
+            t0=t0,
+            max_order=max_order,
+            sigma2_awgn=sigma2_awgn,
+            sources=sources,
+            mics=mics,
+            materials=materials,
+            temperature=temperature,
+            humidity=humidity,
+            air_absorption=air_absorption,
+            ray_tracing=ray_tracing,
+        )
+
+    def __str__(self):
+
+        return "AnechoicRoom instance in {}D.".format(self.dim)
+
+    def is_inside(self, p):
+        """Overloaded function to eliminate testing if objects are inside ``room''."""
+        # always return True because we want the walls to have no effect.
+        return True
```

Comment:
That's an interesting way to get around the size problem 🙈 I had not thought of that... But I like it!

Reply:
Thanks :)